packages/rich-text/src/formatters/color.ts

```diff
@@ -40,11 +42,13 @@ import { isStyle, getVariable } from './util';
  * Color uses PREFIX-XXXXXXXX where XXXXXXXX is the 8 digit
  * hex represenation of the RGBA color.
  */
-const styleToColor = (style) => getSolidFromHex(getVariable(style, COLOR));
+const styleToColor = (style: string): Pattern =>
+  getSolidFromHex(getVariable(style, COLOR));
 
-const colorToStyle = (color) => `${COLOR}-${getHexFromSolid(color)}`;
+const colorToStyle = (color: Solid): string =>
+  `${COLOR}-${getHexFromSolid(color)}`;
 
-function elementToStyle(element) {
+function elementToStyle(element: HTMLElement): string | null {
```

Comment:
(adding this comment here because I can't in `formatters/index.ts`)

I feel like the whole formatters array needs a proper type - something like:

```ts
const formatters: Formatter[] = [
  ...
];
```

That will make using this array a bit cleaner, as it currently is a weird inferred intersection type, that doesn't really guarantee anything about the members in general, only the specific members that are extracted.

The `Formatter` interface would be something like:

```
interface Formatter {
  elementToStyle: (element: HTMLElement) => string | null;
  stylesToCSS: (styles: DraftInlineStyle) => CSSProperties | null;
  getters: Record<string, Getter>;
  setters: Record<string, Setter>;
  autoFocus: boolean;
}
```

This does require extra work of course to make sure it complies with that, but it makes several of the instances of using the `formatters` array a lot cleaner.

Note that the `stylesToCSS` function inside `uppercase.ts` has to be amended to `{ textTransform: 'uppercase' as const }` to pass as a valid CSSProperties object.

Reply:
Added the `Formatter` interface but with `setters` as `Record<string, unknown>`. Couldn't figure out a better type currently since the setters accept different types as args.